src/scripts/Joyride.jsx

```diff
@@ -883,10 +883,17 @@ class Joyride extends React.Component {
     if (/^left/.test(position) && rect.left - (component.width + tooltipOffset) < 0) {
       position = 'top';
     }
-    else if (/^right/.test(position) && (rect.left + rect.width + (component.width + tooltipOffset)) > body.width) {
+    else if (/^right/.test(position) && (rect.left + rect.width + (component.width + tooltipOffset)) > body.getBoundingClientRect().width) {
       position = 'bottom';
     }
 
+    if (/^top/.test(position) && (rect.top + body.scrollTop) - (component.height + tooltipOffset) < 0) {
+      position = 'bottom';
+    }
+    else if (/^bottom/.test(position) && (rect.bottom + body.scrollTop) + (component.height + tooltipOffset) > getDocHeight()) {
+      position = 'top';
+    }
```

Comment:
@gilbarbara I think this part of the PR is not handled so far by your commit, is it?